data_safe_haven/infrastructure/programs/sre/workspaces.py

```diff
@@ -179,11 +195,15 @@ def __init__(
         ]
         for test_file in pathlib.Path(resources_path / "workspace").glob("test*"):
             file_uploads.append((FileReader(test_file), "0444"))
-        for vm, vm_output in zip(vms, vm_outputs, strict=True):
+        for vm_index, (vm, vm_output) in enumerate(
+            zip(vms, vm_outputs, strict=True), start=1
+        ):
             outputs: dict[str, Output[str]] = {}
             for file_upload, file_permissions in file_uploads:
                 file_smoke_test = FileUpload(
-                    replace_separators(f"{self._name}_file_{file_upload.name}", "_"),
+                    replace_separators(
+                        f"workspace_{vm_index:02d}_file_{file_upload.name}", "_"
+                    ),
```

Comment:
In the future it might be better to do this kind of thing with Ansible. Not a change for here though.